src/components/cc-token-oauth-list/cc-token-oauth-list.smart.js

```diff
@@ -0,0 +1,223 @@
+      });
+
+    onEvent(
+      'cc-token-oauth-list:revoke-oauth-token',
+      /** @param {string} tokenId */
+      (tokenId) => {
+        updateOneToken(tokenId, (oauthTokenState) => {
+          oauthTokenState.type = 'revoking';
+        });
+
+        api
+          .revokeOauthToken(tokenId)
+          .then(() => {
+            updateComponent(
+              'state',
+              /** @param {TokenOauthListStateLoaded} state */
+              (state) => {
+                state.oauthTokens = state.oauthTokens.filter((token) => token.id !== tokenId);
+              },
+            );
+            notifySuccess(i18n('cc-token-oauth-list.revoke-token.success'));
+          })
+          .catch((error) => {
+            console.error(error);
+            updateOneToken(tokenId, (oauthTokenState) => {
+              oauthTokenState.type = 'idle';
+            });
+            notifyError(i18n('cc-token-oauth-list.revoke-token.error'));
+          });
+      },
+    );
+
+    onEvent('cc-token-oauth-list:revoke-all-oauth-tokens', () => {
+      updateComponent(
+        'state',
+        /** @param {TokenOauthListStateLoaded} state */
+        (state) =>
+          /** @type {TokenOauthListStateRevokingAll} */ ({
+            ...state,
+            type: 'revoking-all',
+            oauthTokens: state.oauthTokens.map((token) => ({ ...token, type: 'revoking' })),
+          }),
+      );
+
+      const tokens = /** @type {TokenOauthListStateLoaded} */ (component.state).oauthTokens;
+
+      api.revokeAllOauthTokens(tokens).then(({ remainingTokens, errors, revokedTokens }) => {
+        updateComponent('state', (state) => {
+          state.type = 'loaded';
+
+          /** @type {TokenOauthListStateLoaded} */
+          (state).oauthTokens = remainingTokens.map((token) => ({
+            ...token,
+            type: 'idle',
+          }));
+
+          return state;
+        });
+
+        if (errors.length === 0) {
+          notifySuccess(i18n('cc-token-oauth-list.revoke-all-tokens.success'));
+        } else if (revokedTokens.length > 0) {
+          notifyError(i18n('cc-token-oauth-list.revoke-all-tokens.partial-error'));
+        } else {
+          notifyError(i18n('cc-token-oauth-list.revoke-all-tokens.error'));
+        }
+      });
+    });
+  },
+});
+
+class Api {
+  /**
+   * @param {ApiConfig} apiConfig
+   * @param {string} authBridgeConsumerKey
+   */
+  constructor(apiConfig, authBridgeConsumerKey) {
+    this._apiConfig = apiConfig;
+    this._authBridgeConsumerKey = authBridgeConsumerKey;
+  }
+
+  /**
+   * Fetches and formats OAuth tokens
+   *
+   * @returns {Promise<OauthToken[]>} A promise that resolves to an array of formatted OAuth tokens
+   */
+  getOauthTokens() {
+    return getAllTokens()
+      .then(sendToApi({ apiConfig: this._apiConfig }))
+      .then(
+        /** @param {Array<RawTokenData>} tokens */
+        (tokens) => {
+          const filteredAndFormattedTokens = tokens
```

Comment:
nitpick: do we really need this variable?

Reply:
we don't need it anymore since the map callback has been typed :+1: